### PR TITLE
Add `api/v1/info` endpoint for relayer info.

### PR DESCRIPTION
**Pull request by shekohex**
This is the following PR after #3 landed in `main`.
this PR adds the following HTTP endpoint: GET `api/v1/info`
this API Endpoint will respond back with relayer configuration.

**Example**:
```json

{

    "substrate": {
        "webb": { },
        "edgeware": null,
        "hedgeware": null,
        "beresheet": null
    },
    "evm": {
        "webb": null,
        "edgeware": null,
        "ganache": {
            "withdrewFee": "0x5",
            "withdrewGaslimit": "0x350000",
            "account": "0x7b1afe2745533d852d6fd5a677f14c074210d896"
        },
        "hedgeware": null,
        "beresheet": {
            "withdrewFee": "0x5",
            "withdrewGaslimit": "0x350000",
            "account": "0x48d41e139d3133f1879ce5080b9c2cb4878332c2"
        },
        "harmony": {
            "withdrewFee": "0x5",
            "withdrewGaslimit": "0x350000",
            "account": "0x48d41e139d3133f1879ce5080b9c2cb4878332c2"
        }
    }

}

```

**Notes**:

* if the network is `null` that obviously means this network is not supported or not configured correctly for this relayer.
* The `account` is derived from the account's private key, this would be used during zkp as the value of the relayer account.

### CODE REVIEW

**Comment by nepoche** [comment]
Are there any updates that need to be made to the relayer's config.toml? Using postman, I attempted a GET request for `http://localhost:9955/api/v1/info`, but errored out.

**Comment by shekohex** [comment]
Can you please share the error? also, can you try to open the url in the browser too?

**Comment by nepoche** [comment]
Here are some screenshots of the setup and error:
Screen Shot 2021-07-15 at 4 45 23 PM
Screen Shot 2021-07-15 at 4 45 58 PM
Screen Shot 2021-07-15 at 4 46 32 PM

File: src/handler.rs [hunk 1]
```diff
@@ -79,6 +79,39 @@ pub async fn handle_ip_info(
...
+                .$f
+                .as_mut()
+                .zip(ctx.evm_wallet::<$network>().await.ok())
+            {
+                c.account = Some(w.address());
+            }
+        };
+    }
+
+    update_account_for!(webb, evm::Webb);
```
Comment (drewstone) [inline src/handler.rs:106]: cool macro! can you document it a bit more?

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks great!

File: src/handler.rs (hunk 1 above)
Comment (shekohex) [reply src/handler.rs:115]: Done.

File: tests/main.ts [hunk 2]
```diff
@@ -348,6 +348,9 @@ async function main() {
   console.log('Starting the Relayer ..');
   const relayer = await startWebbRelayer();
   await sleep(500); // just to wait for the relayer start-up
+  // get all relayer information
+  const relayerInfoRes = await fetch('http://localhost:9955/api/v1/info');
```
Comment (shekohex) [inline tests/main.ts:352]: @nepoche I added a way to query relayer information during the tests here, so we at least make sure it works

File: tests/main.ts [hunk 3]
```diff
@@ -399,6 +402,8 @@ async function main() {
     contractAddress
   );
   console.log('Proof Generated!');
+  const relayerChainInfo = relayerInfo.evm[testedChain.name];
+  console.log({ relayerChainInfo });
```
Comment (shekohex) [inline tests/main.ts:406]: and reading the `withdrewFee` is easy from here.

File: src/chains.rs [hunk 4]
```diff
@@ -118,7 +118,7 @@ pub mod evm {
     }
 
     define_chain! {
-        Harmoney => {
+        Harmony => {
```
[the remark this answers is not in the document]
Comment (nepoche) [reply src/chains.rs:121]: From what I can tell, the shards operate as separate EVMs. So if we are deploying on multiple shards, then we should define separate chains.

**Pull request merged**